memory-viz/src/memory_model.ts

```diff
@@ -180,19 +180,31 @@ export class MemoryModel {
                 return this.drawDict(x, y, id, value, style);
             } else if (
                 type === "set" &&
-                isArrayOfType<number>(value, "number")
+                (isArrayOfType<number>(value, "number") ||
```

Comment:
This code strikes me as a bit weird; I'm not sure it correctly handles cases when there's a mixture of numbers and `null` values. Please look into this, and add new test cases as needed. You might need to change the implementation of `isArrayOfType`.